wallet/wallet.c

```diff
@@ -403,8 +403,10 @@ static const struct utxo **wallet_select(const tal_t *ctx, struct wallet *w,
 		 * confirmation height and that it is below the required
 		 * maxheight (current_height - minconf) */
 		if (maxheight != 0 &&
-		    (!u->blockheight || *u->blockheight > maxheight))
+		    (!u->blockheight || *u->blockheight > maxheight)) {
+			tal_free(u);
```

Comment:
Doing so will unreserve the utxo, right ? If so, I think that's what I missed to fix #3591 in #3593 !

Reply:
Not yet: we are in `wallet_select` so the coins have not been marked as selected yet (that happens 6 lines down). It's the destructor added to the tallocated array on line 361 that'd free them, but they were not being added yet (just skipped) before this fix.

Reply:
That's a roundabout way of saying that to unmark selected UTXOs you need to `tal_free` the `utxos` array :-)